packages/react-reconciler/src/ReactFiberClassComponent.js

```diff
@@ -908,27 +900,38 @@ export default function(
           capturedValues,
         );
       }
-    } else {
-      newState = oldState;
     }
 
-    if (derivedStateFromProps !== null && derivedStateFromProps !== undefined) {
+    if (derivedStateFromCatch !== null && derivedStateFromCatch !== undefined) {
       // Render-phase updates (like this) should not be added to the update queue,
       // So that multiple render passes do not enqueue multiple updates.
       // Instead, just synchronously merge the returned state into the instance.
       newState =
         newState === null || newState === undefined
-          ? derivedStateFromProps
-          : Object.assign({}, newState, derivedStateFromProps);
+          ? derivedStateFromCatch
+          : Object.assign({}, newState, derivedStateFromCatch);
     }
-    if (derivedStateFromCatch !== null && derivedStateFromCatch !== undefined) {
+
+    let derivedStateFromProps;
+    if (oldProps !== newProps) {
+      // In this case, the prevState parameter should be the partially updated state,
```

Comment:
Conceptually, what you're calling `prevState` is the work-in-progress state. So your comment here applies to all cases; it's not special.

Reply:
"in this case" just referred to the _update_ case. The other two invocations of this method are _mounting_ cases. But I'll re-word the comment.